internal/service/database/properties/schema.go

```diff
@@ -110,127 +121,200 @@ func stringSlice(val interface{}) ([]string, bool) {
+			)
+		}
 
 		if prop.Minimum != nil {
-			validations = append(validations, validation.IntAtLeast(int(*prop.Minimum)))
+			s.Validators = append(s.Validators, int64validator.AtLeast(int64(*prop.Minimum)))
 		}
 		if prop.Maximum != nil {
 			if *prop.Maximum < float64(math.MaxInt) {
-				validations = append(validations, validation.IntAtMost(int(*prop.Maximum)))
+				s.Validators = append(s.Validators, int64validator.AtMost(int64(*prop.Maximum)))
 			}
 		}
-	case "number":
-		s.Type = schema.TypeFloat
+
+		return s, nil
+	case propTypeNumber:
+		s := schema.Float64Attribute{
+			Description:   getDescription(key, prop),
+			Optional:      true,
+			Computed:      true,
+			Sensitive:     isSensitive(key),
+			PlanModifiers: []planmodifier.Float64{},
+			Validators:    []validator.Float64{},
+		}
+
+		if prop.CreateOnly {
+			s.PlanModifiers = append(
+				s.PlanModifiers,
+				float64planmodifier.RequiresReplace(),
+				float64planmodifier.UseStateForUnknown(),
+			)
+		}
 
 		if prop.Minimum != nil {
-			validations = append(validations, validation.FloatAtLeast(*prop.Minimum))
+			s.Validators = append(s.Validators, float64validator.AtLeast(*prop.Minimum))
 		}
 		if prop.Maximum != nil {
-			validations = append(validations, validation.FloatAtMost(*prop.Maximum))
+			s.Validators = append(s.Validators, float64validator.AtMost(*prop.Maximum))
+		}
+
+		return s, nil
+	case propTypeBoolean:
+		s := schema.BoolAttribute{
+			Description:   getDescription(key, prop),
+			Optional:      true,
+			Computed:      true,
+			Sensitive:     isSensitive(key),
+			PlanModifiers: []planmodifier.Bool{},
+		}
+
+		if prop.CreateOnly {
+			s.PlanModifiers = append(
+				s.PlanModifiers,
+				boolplanmodifier.RequiresReplace(),
+				boolplanmodifier.UseStateForUnknown(),
+			)
 		}
-	case "boolean":
-		s.Type = schema.TypeBool
 
 		if boolDefault, ok := booleanDefault(prop.Default); ok {
-			s.Computed = false
-			s.Default = boolDefault
+			s.Default = booldefault.StaticBool(boolDefault)
 		}
-	case "array":
-		s.Type = schema.TypeList
-		s.Elem = &schema.Schema{
-			Type: schema.TypeString,
+
+		return s, nil
+	case propTypeArray:
+		s := schema.ListAttribute{
+			Description:   getDescription(key, prop),
+			ElementType:   types.StringType,
+			Optional:      true,
+			Computed:      true,
+			Sensitive:     isSensitive(key),
+			PlanModifiers: []planmodifier.List{},
+		}
+
+		if prop.CreateOnly {
+			s.PlanModifiers = append(
+				s.PlanModifiers,
+				listplanmodifier.RequiresReplace(),
+				listplanmodifier.UseStateForUnknown(),
+			)
 		}
-	case "object":
-		nested, err := getSchemaMap(prop.Properties)
+
+		return s, nil
+	case propTypeObject:
+		nested, err := getNestedObject(prop.Properties)
 		if err != nil {
 			return nil, err
 		}
 
-		s.Type = schema.TypeList
-		s.MaxItems = 1
-		s.Elem = &schema.Resource{Schema: nested}
-	default:
-		return nil, fmt.Errorf(`unknown property value type "%s"`, prop.Type)
-	}
+		s := schema.ListNestedBlock{
```

Comment:
Figured it would be convenient to use same schema type as before, but looking at the complexity of the value conversions, switching to `SingleNestedBlock` (here and in nested object properties) and adding a `UpgradeState` might have been more simple 🤔